controllers/templatesync/template_sync.go

```diff
@@ -367,6 +367,13 @@ func (r *PolicyReconciler) Reconcile(ctx context.Context, request reconcile.Requ
 			continue
 		}
 
+		// if there's a duplicate name, append the kind since duplicates should only exist for different kinds
+		for _, tmpName := range templateNames {
+			if tmpName == tName {
+				tName = fmt.Sprintf("%s-%s", tName, strings.ToLower(object.GetObjectKind().GroupVersionKind().Kind))
```

Comment:
Let's use an invalid Kubernetes object name format for this. Also, I think you can just use `GetKind()`.

Also, why is the match case insensitve? Is Kubernetes not case sensitive for the `kind` field?
```suggestion
				tName = fmt.Sprintf("%s:%s", tName, strings.ToLower(object.GetKind()))
```

Reply:
Oh, I see. You're actually modifying the template name. I think we should avoid this as this is confusing behavior for the user.

Is the issue that the status sync gets confused for which template is being referenced? If that's the case, then it should refer back to the event.Related field (if present) which says which policy template that status event is for and determine which policy template that way. For the Gatekeeper controller, it always sets it:
https://github.com/open-cluster-management-io/governance-policy-framework-addon/blob/da526413602d46c949a79ef867ee57743a468024/controllers/utils/events.go#L69-L75

Reply:
If it's truly a console issue like you say in the PR description, then we should have a console fix instead.